src/Language/DifferentialDatalog/rust/datalog/test.rs

```diff
@@ -0,0 +1,787 @@
+#![allow(non_camel_case_types)]
+
+use program::*;
+use uint::*;
+use abomonation::Abomonation;
+
+use std::sync::{Arc,Mutex};
+use fnv::FnvHashSet;
+use std::iter::FromIterator;
+
+const TEST_SIZE: u64 = 10000;
+
+#[derive(Eq, Ord, Clone, Hash, PartialEq, PartialOrd, Serialize, Deserialize, Debug)]
+enum Value {
```

Comment:
Why is this only in Test? Isn't this needed elsewhere?

Reply:
The client of the library defines their own value type. It will be different for each Datalog program.